docs/flow.md

```diff
@@ -0,0 +1,33 @@
+# Flow of tasks
+
+Below diagram demonstrates how a workflow is processed by Cadence workers and backend components.
+Note that this is a partial diagram that only highlights selected components and their interactions. There is more happening behind the scenes.
+
+[comment]: <> (To visualize mermaid flowchart below, install Mermaid plugin for your IDE. Works in github out of the box)
+
+```mermaid
+flowchart TD
+    User((Domain X Admin)) -->|"0. Deploy worker with config: domain, tasklist(s)"| Worker[[\nDomain X Worker\n\n]]
+    ExternalInitiator(("Domain X\nAnother Service/CLI")) -->|"1. Start/Signal Workflow"| Frontend
+    Worker -->|2. Poll tasks for domain: X, tasklist: tl1| Frontend[[\nCadence Frontend\n\n]]
+    Frontend -->|3. Forward Poll request| Matching[[\nCadence Matching\n\n]]
+    Matching -->|4. Poll| TaskListManager(Tasklist Manager for tl1)
+    TaskListManager -->|5. Long Poll| WaitUntilTaskFound
+    WaitUntilTaskFound --> |6. Return task to worker| Worker
+    Worker -->|7. Generate decisions\n& respond| Frontend
+    Frontend -->|8. Respond decisions| History[[\nCadence History\n\n]]
+    History -->|9. Mutable state update| ExecutionsTable[(Executions Table)]
+    History -->|10.a. Notify| HistoryQueue(History Queues - per shard)
+    HistoryQueue -->|Periodic task fetch| ExecutionsTable
+    HistoryQueue -->|11.b. Execute transfer task| TransferTaskExecutor(Transfer Task Executor)
+    HistoryQueue -->|11.a. Execute timer task| TimerTaskExecutor(Timer Task Executor)
+    HistoryQueue -->|Periodic per-shard offset save| ShardsTable[(Shards Table)]
```

Comment:
I would maybe put a * here that this is part of the Executions table in Cassandra

Reply:
sounds good. let me clarify below what each of these "tables" actually refer to